sim-lib/src/random_activity.rs

```diff
@@ -220,3 +239,159 @@ impl Display for PaymentActivityGenerator {
         )
     }
 }
+
+#[cfg(test)]
+mod tests {
+    mod test_network_graph_view {
+        use lightning::ln::features::Features;
+        use ntest::timeout;
+
+        use super::super::*;
+        use crate::test_utils::get_random_keypair;
+        use crate::NodeInfo;
+
+        fn create_nodes(n: usize, node_capacity: u64) -> Vec<(NodeInfo, u64)> {
+            (1..=n)
+                .map(|_| {
+                    (
+                        NodeInfo {
+                            pubkey: get_random_keypair().1,
+                            alias: String::new(),
+                            features: Features::empty(),
+                        },
+                        node_capacity,
+                    )
+                })
+                .collect()
+        }
+
+        #[test]
+        fn test_new() {
+            // Check that we need, at least, two nodes
+            for i in 0..2 {
+                assert!(matches!(
+                    NetworkGraphView::new(create_nodes(i, 42 * (i as u64 + 1))),
+                    Err(SimulationError::RandomActivityError { .. })
+                ));
+            }
+
+            // Check that, even if we have two nodes, the the node capacity of all of them must be greater than 0
+            // One of them is 0
+            let mut nodes = create_nodes(1, 0);
+            nodes.extend(create_nodes(1, 0));
+            assert!(matches!(
+                NetworkGraphView::new(nodes),
+                Err(SimulationError::RandomActivityError { .. })
+            ));
+
+            // All of them are 0
+            assert!(matches!(
+                NetworkGraphView::new(create_nodes(2, 0)),
+                Err(SimulationError::RandomActivityError { .. })
```

Comment:
Rust question: should we be: 
1. Returning different enum variants for these errors.
2. String matching the error case to make sure we're covering what we expect.

Not sure which is canonical, but rn it's easy to miss coverage for a branch if the test is changed because we're only asserting "some error happened" here.

Reply:
I think different errors should be better if we want finner granularity here, especially since pattern matching based on error messages is more brittle and will potentially involve larger diff down the line.

If we go with one, I can see at least two categories here that we may want to split:

- `ValueError` referring to one of the input values being invalid. This would apply to the list of provided nodes not being big enough, the source_capacity/expected_payment/multiplier being zero, ...
- `InsufficientCapacity` referring to a node not having enough capacity to process a payment, a `NetworkGraphView` created with zero capacity

This would cover all current cases I think. Do you think we should go even finner?

Reply:
> Do you think we should go even finner?

I think that's fine grained enough, especially since we're only using them in tests 👍

Reply:
I've addressed this in [fd06118](https://github.com/bitcoin-dev-project/sim-ln/pull/143/commits/fd061181ab47ba88c844f1d6de9031a0e65e816b) alongside some error re-definition. Please check whether that makes sense

Reply:
Good enough error update for now 👍  Hopefully as we include tests with new codes we need to do less of these refactor changes.